cmake/modules/FindMySQL.cmake

```diff
@@ -10,11 +10,19 @@ if(MYSQL_INCLUDE_DIR OR MYSQL_)
 endif()
 
 if(NOT WIN32)
-  find_program(MYSQL_CONFIG_EXECUTABLE mysql_config
-    /usr/bin/
-    /usr/local/bin
-    ${MYSQL_DIR}/bin $ENV{MYSQL_DIR}/bin
-  )
+  # Split into two find_program invocations to avoid user-specified
+  # mariadb_config being overridden by system mysql_config.
+  find_program(MYSQL_CONFIG_EXECUTABLE NAMES mysql_config mariadb_config
```

Comment:
This will override what the user sets anyway, no? Only the second way (inside the `if`) should be necessary, so that if `MYSQL_CONFIG_EXECUTABLE` has been set by the user, it won't be overwritten here.

Reply:
Isn't the first one handling the case where the user set MYSQL_DIR but does not set MYSQL_CONFIG_EXECUTABLE?

Reply:
Well, yes, but it's also handling the case when the user does set `MYSQL_CONFIG_EXECUTABLE` by overwriting it.

Reply:
Tests indicate that `find_program` does *not* overwrite the variable if it is already set by the user with `-D`.

Reply:
Yes, [you are right](https://cmake.org/cmake/help/latest/command/find_program.html), it will not overwrite. In any case, I still think we don't need two calls. It should be possible to combine both into a single one that still works as intended.

Reply:
The problem is that the searches are depth-first rather than name-first. That means that `mysql_config` from (_e.g._) `/usr/bin` will be picked up in preference to `mariadb_config` from the place I specify unless I look in _only_ the places I specify first, and then allow "default" areas later.